### ISSUE TITLE
Remote DataSets : add new option to truncate dataSet data even with no new data returned

### ISSUE THREAD

**Issue by PeterMis** [xibosignage/xibo#2620]
In version 2 due to incorrect workflow, regardless of the number of returned entries from the remote dataSet the data was truncated (if truncate was enabled).

In version 3 we've fixed the workflow issue, as such the data will only be truncated when new data is returned (and truncate is enabled).

However, even though the workflow is correct in version 3, some users may not expect this change, as such it might be desirable to implement a new option to remote DataSet configuration that would allow for it to be truncated regardless if new data is returned or not.

**Comment by Commifreak** [comment]
Sorry to bother you - but since the upgrade an enabling this feature, I loose every data in dataset.

I enabled debug and catched:

```
Making request to http://xx/XiboOutputV2.php with params: array ( 'query' => array ( 'planned' => '', ), )
Cache Control Key is d751713988987e9331980363e24189ce
Skipping due to MD5
Truncate Raumverwaltung Geplant PROD
```

So - its skipping due to MD% and truncate the data? And I see that data IS returned (even on the test page).

Is there an issue between caching and truncating?

Testing this datasource:

```
{
   "entries": [
      [
** data sanitized **
      ]
   ],
   "number": 4,
   "processed": [
      [
**data sanitized **
      ]
   ],
   "messages": [
      "Verarbeiten 1 Ergebnisse in 6 potenziellen Spalten",
      "Verarbeitungsergebnis mit Datenstamm ",
      "DataRoot is an array",
      "Processing 0",
      "Processing 1",
      "Processing 2",
      "Processing 3",
      "Konsolidieren von Einträgen",
      "Es gibt insgesamt 4 Einträge"
   ]
}
```

Or should I continue discuss at https://community.xibo.org.uk/t/datasets-truncate-even-if-no-data-is-returned/25581/4 ?

**Comment by dasgarner** [comment]
@PeterMis please can you take a look at this for @Commifreak ? I don't think we should truncate if we've used cached data, right?

**Comment by Commifreak** [comment]
Hi,

I got one workaround for one system but cannot set this in the second, so it would be great if there would be a native solution for this :)

To workaround I currently always send unique data to xibo, so the checksum is always different, although the end user does not see any difference viewirng the data.

**Comment by dasgarner** [comment]
Sorry we've been super busy over the last few weeks - and years really :smile: 

I did grab a moment to look at this and I think I see whats happening. If you wanted to try a patch I think this will do it: https://github.com/dasgarner/xibo-cms/commit/c6207c876208c3a192dedbf3c42ee24b4ee9f2e0

**Comment by Commifreak** [comment]
Tested it and: It works now!

**Comment by dasgarner** [comment]
Thanks for testing it, i'm pleased its working.

Yes the Linux player is more like an epic

**Comment by Commifreak** [comment]
This fix will be included in 3.1.2 then?

**Comment by dasgarner** [comment]
Yes it will be - see the linked issue for the milestone tracking.
